File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.c
int smallestIndex(int* nums, int numsSize) {
        for(int i=0;i<numsSize;i++)
        {
            int sum=0;
            int n=nums[i];
            while(n>0)
            {
                int dig=n%10;
                sum+=dig;
                n=n/10;
            }
            if(sum==i)
            {
                return i;
            }
        }
        return -1;
}

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            sum=0
            n=nums[i]
            while(n>0):
                dig=n%10
                sum+=dig
                n=n//10
            if(sum==i):
                return i
        return -1